XenModel/XenAPI-Extensions/VM.cs

```diff
@@ -756,9 +790,13 @@ public VirtualisationStatus GetVirtualisationStatus
                     if (vm_guest_metrics != null && vm_guest_metrics.PV_drivers_installed)
                     {
                         if (vm_guest_metrics.PV_drivers_up_to_date)
-                            return VirtualisationStatus.OPTIMIZED;
+                        {
+                            return VirtualisationStatus.OPTIMIZED | VirtualisationStatus.IO_DRIVERS_INSTALLED | VirtualisationStatus.MANAGEMENT_INSTALLED;

```

Comment:
This seems wrong, because it's not looking up GetNewVirtualisationFlags for new VMs.